src/coreclr/jit/helperexpansion.cpp

```diff
@@ -478,36 +478,49 @@ bool Compiler::fgExpandThreadLocalAccessForCall(BasicBlock** pBlock, Statement*
         return false;
     }
 
+    assert(!opts.IsReadyToRun());
```

Comment:
Unrelated to this line:

Can you fix `doesMethodHasTlsFieldAccess()` to have proper grammar? Change it to just `methodHasTlsFieldAccess()`.

(With `does`, it would need to be `doesMethodHaveTlsFieldAccess()`, but then `if (doesMethodHaveTlsFieldAccess())` reads really awkwardly)

Reply:
There are quite a lot methods that has `doesMethod*`, but I will leave that as a cleanup for some different PR. Here, I updated with your suggestion.